### PR TITLE
Release notes 5.0

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

This PR updates the changelog with 5.0 release content.

## Related Issue

Closes PWA-252

## Acceptance 

@jimbo
@awilcoxa

### Verification Stakeholders

@jimbo to verify items and descriptions
@awilcoxa to verify items and descriptions

### Verification Steps

1. Open `CHANGELOG.md`
2. Verify items and descriptions are accurate

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

**Comment by dpatil-magento** [comment]
@jcalcaben -  These PR's are already part of v4.0.0 (https://github.com/magento/pwa-studio/commits/v4.0.0) , you can remove them -
https://github.com/magento/pwa-studio/pull/1640
https://github.com/magento/pwa-studio/pull/1575
https://github.com/magento/pwa-studio/pull/1584
https://github.com/magento/pwa-studio/pull/1637

**Comment by jcalcaben** [comment]
@dpatil-magento Entries removed

**Comment by revanth0212** [comment]
Can you add #2093

**Comment by dpatil-magento** [comment]
Thanks @revanth0212 ,  @jcalcaben OR you can add actual merged one https://github.com/magento/pwa-studio/pull/2086.

Below are some more missing ones to add, which are merged recently.
https://github.com/magento/pwa-studio/pull/2056
https://github.com/magento/pwa-studio/pull/2054
https://github.com/magento/pwa-studio/pull/2085
https://github.com/magento/pwa-studio/pull/2055

**Comment by jcalcaben** [comment]
@dpatil-magento Entries added

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,250 +1,310 @@
...
+This release makes the Venia `RichContent` component compatible with the default Page Builder content types.
 
--   [Toast hooks][] and [ToastContainer][] - Provide logic for managing toast messages and a container for displaying these messages.
--   [`useWindowSize()`][] - A hook that provides window size data and lets you respond to window resizing events.
--   [`useRestApi()`][] and [`useRestResponse()`][] - Hooks that expose the API for sending REST calls and handling the response.
--   [`useScrollLock()`][] - Hook that gives the ability to lock the `document` element.
--   [`usePagination()`][] - A hook that provides pagination logic for components that need to navigate through paged data.
--   [`useEventListener()`][] - A hook that gives the ability to add a callback function when an event is triggered on an element.
--   [`useCarousel()`][] - A hook for interacting with the state for a carousel of images.
+### Scaffolding
```
Comment (cherdman) [inline CHANGELOG.md]: I would link to the additional docs on Scaffolding.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,250 +1,310 @@
...
+This command is a user-friendly version of the `create-project` sub-command in the `pwa-buildpack` CLI tool.
 
-#### Component refactors
+Use this command to set up an initial project structure using the Venia storefront as a template.
+It provides an interactive questionnaire to help configure the different parts of your project.
 
-This release includes code refactors to existing components.
-The main purpose for this refactor is to extract the logic from these components and convert them into re-useable Peregrine hooks, such as `useCarousel()` and `useScrollLock()`.
-Another reason for refactoring some of these components is to convert them into functional components.
+### Talons
```
Comment (cherdman) [inline CHANGELOG.md]: I would link to additional docs on Talons and Peregrine.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,250 +1,310 @@
...
-The `venia-ui` package contains the templates and components used to create the Venia storefront.
-This package is meant to be used as a component library for custom storefront projects.
+This release refactors the way PWA Studio handles state management in Venia.
+Instead of using Redux directly, Peregrine now provides a set of context providers and hooks to interact with the different slices of state in the global data store.
 
-The `venia-concept` package contains the project scripts and configurations used to build and run the actual Venia storefront project.
+For more information, read [State management](https://pwastudio.io/technologies/basic-concepts/state-management/)
 
-This change may also require you to update import paths in your project.
+### REST to GraphQL migration
```
Comment (cherdman) [inline CHANGELOG.md]: Let's link to the GraphQL devdocs somewhere in here.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,250 +1,310 @@
...
 
-**`babel-preset-peregrine`** - A new package that defines a preset for Babel.
-This preset is required for storefronts using Peregrine and Venia-UI components.
-See PR [#1404][].
+Handling images on the storefront has been improved in this release.
+Storefronts are now able to load the optimal image for a given viewport.
+
+New features such as pre-fetching and lazy loading also boosts page load performance.
+
+### Refactoring classes to functional components
```
Comment (cherdman) [inline CHANGELOG.md]: Do we have any link-able docs relating to our refactor? Maybe a Jimmy-created Gist or something?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,250 +1,310 @@
...
+
+Since this is a major release, some of the changes previously listed may break projects dependent on PWA Studio and its tools and components.
+
+These changes include:
+
+- Refactoring to produce talons have modified the public API of some Venia components
+- Converting from REST to GraphQL calls
+- Optimizing images required updates that modify how images should be used
+- Converting classes to functional components to use React hooks
+- Replacing `MagentoRoutHandler` with new component and talon
```
Comment (cherdman) [inline CHANGELOG.md]: Should read: `MagentoRouteHandler`

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,250 +1,310 @@
...
+| Implemented the `apollo-link-retry` Link                                                    | **Feature**  | [#1799][] |
+| Created component for showing categories with no products                                   | **Feature**  | [#1496][] |
+| Updated template HTML to include noscript fallback                                          |  **Update**  | [#2034][] |
+| Removed id from `getCustomer` response object                                               |  **Update**  | [#2028][] |
+| Added Email validation in signin form                                                       |  **Update**  | [#1981][] |
+| Added support for Carousel appearance for Product content type                              |  **Update**  | [#1951][] |
+| Updated static to venia-static                                                              |  **Update**  | [#1932][] |
+| Upgraded Apollo View layer to hooks                                                         |  **Update**  | [#1827][] |
+| Improved efficienct of the `getNavigationMenu` query                                        |  **Update**  | [#1867][] |
```
Comment (cherdman) [inline CHANGELOG.md]: Should read: "Improved efficiency of the `getNavigationMenu` query"

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,250 +1,310 @@
...
+| Added field labels for inputs                                                               |  **Update**  | [#1845][] |
+| Update mixin references to talon                                                            |  **Update**  | [#1820][] |
```
Comment (cherdman) [inline CHANGELOG.md]: Do we capitalize `Talon` since it is a proper noun? Is it a proper noun? I fail at grammar sometimes...

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,250 +1,310 @@
...
+| Replaced connected components with context hooks                                            |  **Update**  | [#1664][] |
+| Removed sticky attribute from the pagination                                                |  **Update**  | [#1735][] |
+| Replaced `rem` to pixels in the `height` and `width` attribute for the pagination image     |  **Update**  | [#1738][] |
+| Refactored checkout workflow to set shipping address/get shipping methods with GraphQL      | **Refactor** | [#2018][] |
+| Refactored code to use `updateItem` mutations                                               | **Refactor** | [#2017][] |
+| Refactored code to use GraphQL for adding items to cart                                     | **Refactor** | [#1987][] |
+| Replaced REST endpoing for cart item removal with a `removeItemFromCart` mutation           | **Refactor** | [#2015][] |
+| Refactor code to revoke customer token using graphql mutation on sign out                   | **Refactor** | [#2012][] |
+| Refactor code to use GraphQl mutation to create/fetch cartId.                               | **Refactor** | [#1988][] |
```
Comment (cherdman) [inline CHANGELOG.md]: `GraphQl` --> `GraphQL`

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,250 +1,310 @@
...
+| Removed `test.only` from test file                                                          | **Refactor** | [#1989][] |
+| Refactor components to fetch Countries from GraphQL instead of REST                         | **Refactor** | [#1993][] |
+| Replaced sign-in REST with GraphQL                                                          | **Refactor** | [#1904][] |
+| Standardized GQL files with 4-space indentation                                             | **Refactor** | [#1914][] |
```
Comment (cherdman) [inline CHANGELOG.md]: `GQL`--> `GraphQL`

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,250 +1,310 @@
...
+| Added support for initial selections on the **Edit Item** screen                            |  **Update**  | [#1592][] |
+| Replaced connected components with context hooks                                            |  **Update**  | [#1664][] |
+| Removed sticky attribute from the pagination                                                |  **Update**  | [#1735][] |
+| Replaced `rem` to pixels in the `height` and `width` attribute for the pagination image     |  **Update**  | [#1738][] |
+| Refactored checkout workflow to set shipping address/get shipping methods with GraphQL      | **Refactor** | [#2018][] |
+| Refactored code to use `updateItem` mutations                                               | **Refactor** | [#2017][] |
+| Refactored code to use GraphQL for adding items to cart                                     | **Refactor** | [#1987][] |
+| Replaced REST endpoing for cart item removal with a `removeItemFromCart` mutation           | **Refactor** | [#2015][] |
+| Refactor code to revoke customer token using graphql mutation on sign out                   | **Refactor** | [#2012][] |
```
Comment (cherdman) [inline CHANGELOG.md]: `graphql` --> `GraphQL`

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,250 +1,310 @@
...
+| Refactor Gallery component                                                                  | **Refactor** | [#1791][] |
+| Fixed a bug that caused an incorrect default swatch image ratio on the product details page |  **Bugfix**  | [#2055][] |
+| Fixed Search Result counts mismatch                                                         |  **Bugfix**  | [#2037][] |
+| Fixed bug that incorrectly toggled the cart drawer inside the update item action            |  **Bugfix**  | [#2038][] |
+| Fixed bug that caused an error when adding, editing, then saving a configurable item        |  **Bugfix**  | [#2040][] |
+| Fixed Footer position bug which broke z-index styling on Edge                               |  **Bugfix**  | [#2014][] |
+| Fixed storybook                                                                             |  **Bugfix**  | [#1945][] |
+| Fixed invalud default initial value for checkout form                                       |  **Bugfix**  | [#1983][] |
+| Fixed bug where Product price does not change wheb changing configurable options            |  **Bugfix**  | [#1964][] |
```
Comment (cherdman) [inline CHANGELOG.md]: `wheb` --> `when`

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,250 +1,310 @@
...
-| Refactored list components to use hooks                                    | **Refactor** | [#1515][] |
-| Fixed a bug that caused a pagination double fetch                          |  **Bugfix**  | [#1381][] |
-| Fixed peregrine watcher and watch:all                                      |  **Bugfix**  | [#1378][] |
-| Fixed ApolloContext hook                                                   |  **Bugfix**  | [#1467][] |
-| Fixed bug in MagentoRouteHandler so it can correctly handle unknown routes |  **Bugfix**  | [#1495][] |
+| Description                                                                                   | Change type  |    PR     |
+| :-------------------------------------------------------------------------------------------- | :----------: | :-------: |
+| Implemented Product Images pre-fetching                                                       | **Feature**  | [#1938][] |
+| Created a talon for the Search root component                                                 | **Feature**  | [#1953][] |
+| Add ability to lazy load Images                                                               | **Feature**  | [#1871][] |
```
Comment (cherdman) [inline CHANGELOG.md]: `Images` --> `images`

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,250 +1,310 @@
...
-| Refactored the image carousel component into hooks and visual components   | **Refactor** | [#1268][] |
-| Refactored list components to use hooks                                    | **Refactor** | [#1515][] |
-| Fixed a bug that caused a pagination double fetch                          |  **Bugfix**  | [#1381][] |
-| Fixed peregrine watcher and watch:all                                      |  **Bugfix**  | [#1378][] |
-| Fixed ApolloContext hook                                                   |  **Bugfix**  | [#1467][] |
-| Fixed bug in MagentoRouteHandler so it can correctly handle unknown routes |  **Bugfix**  | [#1495][] |
+| Description                                                                                   | Change type  |    PR     |
+| :-------------------------------------------------------------------------------------------- | :----------: | :-------: |
+| Implemented Product Images pre-fetching                                                       | **Feature**  | [#1938][] |
+| Created a talon for the Search root component                                                 | **Feature**  | [#1953][] |
```
Comment (cherdman) [inline CHANGELOG.md]: Should this be consistent with the other talons here? For example, should this instead read: "Created `<name of search root component>` talon"

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,250 +1,310 @@
...
+| Added instructions to circumvent self-signed ssl-certificate errors        |    **Update**     | [#1923][] |
+| Fixed tools and library graphic                                            |    **Update**     | [#1889][] |
+| Updated Venia setup doc                                                    |    **Update**     | [#1809][] |
+| Updated README file                                                        |    **Update**     | [#1717][] |
+| Update contribution guide                                                  |    **Update**     | [#1651][] |
+| Refactored site content navigation                                         |   **Refactor**    | [#1933][] |
+| Refactored doc scripts to remove unused imports and ignore some parameters |   **Refactor**    | [#1835][] |
+| Fixed typo on cloud deployment topic                                       |    **Bugfix**     | [#2007][] |
+| Fixed typo on cloud deployment topic                                       |    **Bugfix**     | [#1974][] |
+| Removed duplicate words from sentences.                                    |    **Bugfix**     | [#1893][] |
```
Comment (cherdman) [inline CHANGELOG.md]: I would eliminate the period to be consistent with the other entries.

File: CHANGELOG.md [hunk 15]
```diff
@@ -278,132 +338,198 @@ This is the easiest way to work with the released versions of PWA Studio.
 
 #### Upgrade method: Update `package.json`
 
-To upgrade to 4.0.0, update the project's `package.json` file and change the dependency version for PWA Studio.
+To upgrade to 5.0.0, update the project's `package.json` file and change the dependency version for PWA Studio.
+
+## Known issues
```
Comment (cherdman) [inline CHANGELOG.md]: Do we not have more Known Issues to call out? @jimbo @dpatil-magento  please weigh in.

Comment (dpatil-magento) [reply CHANGELOG.md]: @jcalcaben  There is one more, might not specific to 5.0 but visible on local development, where images does not load unless you set NODE_TLS_REJECT_UNAUTHORIZED , #2005 would be solving this issue in future.

Comment (jcalcaben) [reply CHANGELOG.md]: @dpatil-magento Added in bf35d48

**Pull request merged**